src/NumericalAlgorithms/Integration/GslQuadAdaptive.cpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Integration/GslQuadAdaptive.hpp"
+
+#include <cstddef>
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_integration.h>
+#include <memory>
+#include <pup.h>
+
+#include "ErrorHandling/Exceptions.hpp"
+#include "Utilities/MakeString.hpp"
+
+namespace integration {
+
+namespace detail {
+
+GslQuadAdaptiveImpl::GslQuadAdaptiveImpl(const size_t max_intervals) noexcept
+    : max_intervals_(max_intervals) {
+  initialize();
+}
+
+void GslQuadAdaptiveImpl::pup(PUP::er& p) noexcept {
+  p | max_intervals_;
+  if (p.isUnpacking()) {
+    initialize();
+  }
+}
+
+void GslQuadAdaptiveImpl::GslIntegrationWorkspaceDeleter::operator()(
+    gsl_integration_workspace* workspace) const noexcept {
+  gsl_integration_workspace_free(workspace);
+}
+
+void GslQuadAdaptiveImpl::initialize() noexcept {
```

Comment:
to resolve one of the clang-tidy issues, default-construct the `gsl_integrand_` here:

```cpp
gsl_integrand_ = gsl_function{};
```